### PR TITLE
DYN-5918 Obsolete Google Analytics

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Obsolete Google Analytics mention and APIs in Dynamo code base, about box.

- [x] Removed GA data collecting consent
- [x] Removed Learn More button
- [x] Reduced overall height of the dialog
- [x] Obsoleted public APIs affected, removed private APIs

image

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

Obsolete Google Analytics mention and APIs in Dynamo code base, about box.

### Reviewers

@DynamoDS/dynamo 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/DynamoCore/DynamoCore.csproj [hunk 1]
```diff
@@ -149,7 +149,9 @@
     <Copy SourceFiles="$(SolutionDir)..\README.md" DestinationFiles="$(OutputPath)README.txt" />
     <Copy SourceFiles="$(SolutionDir)..\doc\distrib\License.rtf" DestinationFolder="$(OutputPath)" />
     <Copy SourceFiles="$(SolutionDir)..\doc\distrib\TermsOfUse.rtf" DestinationFolder="$(OutputPath)" />
+    <!--TODO: Remove in Dynamo 3.0-->
```
Comment (aparajit-pratap) [inline src/DynamoCore/DynamoCore.csproj:152]: Why not remove it now and add it to `legacy_remove_me` so we remember we have to remove this later?

**Review by pinzart90 — APPROVED** [review]
LGTM as long as we do not break the patch installer
We will probably need CSharpAnalytics,Net45.dll and Analytics.NET.Google.dll to be copied over

**Comment by mjkkirschner** [comment]
> LGTM as long as we do not break the patch installer We will probably need CSharpAnalytics,Net45.dll and Analytics.NET.Google.dll to be copied over

could leave this for 3.0 as well?

**Comment by QilongTang** [comment]
> > LGTM as long as we do not break the patch installer We will probably need CSharpAnalytics,Net45.dll and Analytics.NET.Google.dll to be copied over
> 
> could leave this for 3.0 as well?

We could, but I am a bit worried since we already stopped collecting GA data on master now

**Comment by QilongTang** [comment]
There are three regressions on this PR but all of them are related to a setting/ API which will be obsoleted in Dynamo 3.0:
[Dynamo.Tests.PreferenceSettingTests.AnalyticsReportingApprovedSetting](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/10533/testReport/junit/Dynamo.Tests/PreferenceSettingTests/AnalyticsReportingApprovedSetting/)
[DynamoCoreWpfTests.CoreUserInterfaceTests.PreferenceSetting_NotAgreeAnalyticsSharing](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/10533/testReport/junit/DynamoCoreWpfTests/CoreUserInterfaceTests/PreferenceSetting_NotAgreeAnalyticsSharing/)
[Dynamo.Tests.PreferenceSettingTests.AnalyticsReportingApprovedSetting](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/10533/testReport/junit/Dynamo.Tests/PreferenceSettingTests/AnalyticsReportingApprovedSetting_2/)

I expect these tests will be either removed or refactored for Dynamo 3.0

**Comment by aparajit-pratap** [comment]
> LGTM as long as we do not break the patch installer We will probably need CSharpAnalytics,Net45.dll and Analytics.NET.Google.dll to be copied over

This has already been added to legacy files.

**Comment by aparajit-pratap** [comment]
> > > LGTM as long as we do not break the patch installer We will probably need CSharpAnalytics,Net45.dll and Analytics.NET.Google.dll to be copied over
> > 
> > 
> > could leave this for 3.0 as well?
> 
> We could, but I am a bit worried since we already stopped collecting GA data on `master` now. After some thoughts I think it is safer to do so in 3.0.
> 
> @mjkkirschner Do you know when the release branch will be cut?

Why wait for 3.0 when 2.19 already consumes Analytics.NET without GA?

**Comment by QilongTang** [comment]
> ⚠️ **[run-bin-diff-net60-windows]** - Files Added/Deleted::1 file(s) have been deleted! (Updated: 2023-08-01-23:41:23)

hi @zeusongit I saw this warning but I think my commit removed the copy step but added the rtf file to legacy folder to copy back. May I ignore this warning?

**Pull request merged**